static/js/stream_create.js

```diff
@@ -119,45 +119,74 @@ const stream_name_error = new StreamNameError();
-    const $announce_stream_checkbox = $("#announce-new-stream input");
-    const $announce_stream_label = $("#announce-new-stream");
-    let disable_it = false;
+function update_setting_state_based_on_stream_privacy($elem_label, elem_previous_value) {
+    // If the stream is invite only, disable the "Announce stream" and
+    // "Default stream" options. Otherwise enable them.
+    const $elem_checkbox = $elem_label.find("input");
     const privacy_type = $("input:radio[name=privacy]:checked").val();
     const is_invite_only =
         privacy_type === "invite-only" || privacy_type === "invite-only-public-history";
-    $announce_stream_label.removeClass("control-label-disabled");
+    let disable_it = false;
+    $elem_label.removeClass("control-label-disabled");
 
-    // Here, we arrange to save the state of the announce checkbox
-    // when switching to creating a private stream; we will restore it
-    // when switching back to a public stream. This input-disabled
-    // check prevents overwriting stream_announce_previous_value with
+    // Here, we arrange to save the state of the announce and default
+    // stream checkbox when switching to creating a private stream;
+    // we will restore it when switching back to a public stream.
+    // This input-disabled check prevents overwriting previous_value with
     // the false when switching between private stream types.
-    if (!$announce_stream_checkbox.prop("disabled")) {
-        stream_announce_previous_value = $announce_stream_checkbox.prop("checked");
+    if (!$elem_checkbox.prop("disabled")) {
+        elem_previous_value = $elem_checkbox.prop("checked");
     }
 
     if (is_invite_only) {
         disable_it = true;
-        $announce_stream_checkbox.prop("checked", false);
-        $announce_stream_label.addClass("control-label-disabled");
+        $elem_checkbox.prop("checked", false);
+        $elem_label.addClass("control-label-disabled");
     } else {
         // If the stream was already public, this will be a noop.
-        $announce_stream_checkbox.prop("checked", stream_announce_previous_value);
+        $elem_checkbox.prop("checked", elem_previous_value);
+    }
+
+    $elem_checkbox.prop("disabled", disable_it);
+    $elem_label.show();
+
+    return elem_previous_value;
+}
+
+// Within the new stream modal...
+function update_announce_stream_state() {
+    const $announce_stream = $("#announce-new-stream");
+
+    // If there is no notifications_stream, we simply hide the
+    // "Announce stream" widget.
+    if (!stream_data.realm_has_notifications_stream()) {
+        $announce_stream.hide();
+        return;
+    }
+
+    // Update "Announce stream" state
+    stream_announce_previous_value = update_setting_state_based_on_stream_privacy(
+        $announce_stream,
+        stream_announce_previous_value,
+    );
+}
+
+function update_default_stream_state() {
+    const $default_stream = $("#stream_creation_form #default-stream");
```

Comment:
`$("#default-stream")` should be good enough, since IDs are anyways unique to an element. And at a time only one of stream creation or stream edit UI can be opened.

Reply:
I'm afraid of making this change resulting in the same kind of bug as #24238. At a time, only one stream creation or stream edit UI is displayed other is hidden. But we can still select the hidden element. After switching from stream editing UI to stream creation UI using the `$("#default-stream")` selector checkbox of stream editing UI  is being selected.

https://user-images.githubusercontent.com/87542880/221297785-df2838b0-1f54-4425-96f1-3c6d777eef17.mp4

Reply:
@HemantUmre12 generally it's best to leave comments like this left unresolved for the reviewer (in this case @sahil839 to mark as resolved once they are satisfied).